assets/javascripts/needlediff.js

```diff
@@ -138,7 +133,40 @@ NeedleDiff.prototype.draw = function () {
         if (!this.fullNeedleImg) {
           // draw matching part of needle image
           this.ctx.strokeStyle = NeedleDiff.strokecolor(a.type);
-          this.ctx.drawImage(this.needleImg, orig.xpos, orig.ypos, usedWith, a.height, x, a.ypos, usedWith, a.height);
+          if (this.needleImg) {
+            this.ctx.drawImage(this.needleImg, orig.xpos, orig.ypos, usedWith, a.height, x, a.ypos, usedWith, a.height);
+          } else {
```

Comment:
This whole else-block should better be extracted into a separate function (to avoid this one from becoming to big).